frontend/app/components/redesign/components/LinkTagGenerator.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState } from 'react'
+import { cx } from 'class-variance-authority'
+import { Heading5 } from '../Typography'
+import { InputField, ToolsPrimaryButton } from './index'
+import { SVGCopyIcon, SVGCheckIcon } from '~/assets/svg'
+
+export const LinkTagGenerator = () => {
+  const [pointerInput, setPointerInput] = useState('')
+  const [linkTag, setParsedLinkTag] = useState('')
+  const [invalidUrl, setInvalidUrl] = useState(false)
+  const [showCodeBox, setShowCodeBox] = useState(false)
+  const [isCopied, setIsCopied] = useState(false)
+
+  const isValidPointer = (input: string) => {
+    try {
+      let urlString = input.trim()
+      if (input.charAt(0) === '$') {
+        urlString = input.replace('$', 'https://')
+      }
+
+      const url = new URL(urlString)
+
+      if (url.pathname === '/') {
+        return `${url.origin}.well-known/pay`
+      }
+
+      return url.origin + url.pathname
+    } catch {
+      setInvalidUrl(true)
```

Comment:
Should return falsy here, then we can move `isValidPointer` out of this component for better performance, and call `setInvalidUrl` at call-site like rest of hooks.